modules/core/src/main/java/org/apache/synapse/mediators/elementary/Target.java

```diff
@@ -497,6 +497,25 @@ public void removeJson(MessageContext synCtx, SynapsePath jsonPath) throws IOExc
         }
     }
 
+    /**
+     * Renames a json key name at the specified json path with a new key name
+     *
+     * @param synapseContext Current message context
+     * @param jsonPath       The path to locate the key. Should be resolved to a map or an array including map items.
+     * @param keyName        Current name of the key
+     * @param newKeyName     New name of the key
+     * @throws IOException
+     */
+    public void renameKey(MessageContext synapseContext, String jsonPath, String keyName, String newKeyName)
+            throws IOException {
+
+        org.apache.axis2.context.MessageContext axis2MessageCtx =
+                ((Axis2MessageContext) synapseContext).getAxis2MessageContext();
+        String payload = JsonUtil.jsonPayloadToString(((Axis2MessageContext) synapseContext).getAxis2MessageContext());
+        DocumentContext doc = JsonPath.parse(payload).renameKey(jsonPath, keyName, newKeyName);
```

Comment:
Do we have to get the string and then again parse it? Can't we treat it as json

Reply:
`renameKey` method has to be executed on an object of type `DocumentContext`, hence we will have to parse it.